examples/graphics/point-cloud-simulation.html

```diff
@@ -0,0 +1,209 @@
+
+        // generate random positions and old positions within small cube (delta between them represents velocity)
+        for (var i = 0; i < 3 * self.maxNumPoints; i++)
+        {
+            self.positions[i] = Math.random() * 2 - 1;
+            self.oldPositions[i] = self.positions[i] + Math.random() * 0.04 - 0.01;
+        }
+
+        // helper function to update vertex of the mesh
+        function updateMesh(mesh) {
+
+            // Set current positions on mesh - this reallocates vertex buffer if more space is needed to test it.
+            // For best performance, we could preallocate enough space using mesh.Clear.
+            // Also turn off bounding box generation, as we set up large box manually
+            mesh.setPositions(this.positions, 3, visiblePoints);
+            mesh.update(pc.PRIMITIVE_POINTS, false);
+        }
+
+        // Create a mesh with dynamic vertex buffer (index buffer is not needed)
+        this.mesh = new pc.Mesh(this.app.graphicsDevice);
+        this.mesh.clear(true);
+        updateMesh(this.mesh);
+
+        // set large bounding box so we don't need to update it each frame
+        this.mesh.aabb = new pc.BoundingBox(new pc.Vec3(0, 0, 0), new pc.Vec3(15, 15, 15));
+
+        // Create the shader from the vertex and fragment shaders
+        shader = new pc.Shader(app.graphicsDevice, {
+            attributes: { aPosition: pc.SEMANTIC_POSITION },
+            vshader: document.getElementById("vshader").textContent,
+            fshader: document.getElementById("fshader").textContent
+        });
+
+        // Create a new material with the new shader and additive alpha blending
+        material = new pc.Material();
+        material.shader = shader;
+        material.blendType = pc.BLEND_ADDITIVEALPHA;
+        material.depthWrite = false; 
+
+        // Create the mesh instance
+        var node = new pc.GraphNode();
+        var meshInstance = new pc.MeshInstance(node, this.mesh, material);
+
+        // Create a model and add the mesh instance to it
+        var model = new pc.Model();
+        model.graph = node;
+        model.meshInstances = [ meshInstance ];
+
+        // Create Entity and add it to the scene
+        this.entity = new pc.Entity();
+        app.root.addChild(this.entity);
+
+        // Add a model compoonent
+        app.systems.model.addComponent(this.entity, {
+            type: 'asset',
+            castShadows: false
+        });
+        this.entity.model.model = model;
+
+
+        // Set an update function on the app's update event
+        var self = this;
+        var time = 0, previousTime;
+        app.on("update", function (dt) {
+            previousTime = time;
+            time += dt;
+
+            // update particle positions using simple verlet integration, and keep them inside a sphere boundary
+            var dist, pos = new pc.Vec3(), old = new pc.Vec3(); delta = new pc.Vec3(); next = new pc.Vec3();
```

Comment:
Are `delta` and `next` declared anywhere? Maybe best to do so on separate lines?

Reply:
separated, definitely cleaner.